locator/token_metadata.cc

```diff
@@ -60,7 +60,19 @@ class token_metadata_impl final {
     std::unordered_map<inet_address, inet_address> _replacing_endpoints;
 
     using ring_mapping = boost::icl::interval_map<token, std::unordered_set<inet_address>>;
-    std::unordered_map<sstring, ring_mapping> _pending_ranges_interval_map;
+    // For each keyspace, migration_info contains ranges of tokens and
+    // corresponding replicas to which writes or reads will be directed:
+    //   - pending_endpoints - will be appended to normal endpoints for writes;
+    //   - read_endpoints - will completely replace normal endpoints for reads.
```

Comment:
Why do we need to distinguish read_endpoints?

Reply:
read_endpoints is not set in this commit. If it's set later, it would be better to introduce it then.

Reply:
> Why do we need to distinguish read_endpoints?

`pending_endpoints` are set at `write_both_read_old`, they can only be appended to the normal endpoints, and they have some usages besides just being appended to normal endpoints: https://github.com/scylladb/scylladb/blob/master/service/storage_proxy.cc#L1221, 
https://github.com/scylladb/scylladb/blob/master/service/storage_proxy.cc#L1599
https://github.com/scylladb/scylladb/blob/master/service/storage_proxy.cc#L3626

`read_endpoints` on the other hand are set at `write_both_read_new` and they can skip some endpoints from `normal_endpoints`, that's why they completely replace them.

Reply:
> read_endpoints is not set in this commit. If it's set later, it would be better to introduce it then.

ok

Reply:
> If it's set later, it would be better to introduce it then.

No need to do it now, just for the future.

Reply:
> No need to do it now, just for the future.

already updated